### PR TITLE
fix: revive IsQuorumTypeEnabled logic dropped in 5790

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
`develop` can't sync from genesis on mainnet, https://github.com/dashpay/dash/pull/5790/commits/b8a086d5e7d6c0307f429e3844e4331ffb036c89 broke it.

#5790 follow-up

## What was done?
Revive the old logic but using hardcoded block heights instead of scanning via quorum manager.

## How Has This Been Tested?
Synced on mainnet/testnet, CI is happy https://gitlab.com/UdjinM6/dash/-/pipelines/1148980046.

## Breaking Changes
n/a

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

File: src/llmq/options.cpp [hunk 1]
```diff
@@ -116,28 +116,26 @@ std::map<Consensus::LLMQType, QvvecSyncMode> GetEnabledQuorumVvecSyncEntries()
...
+    const bool fHaveDIP0024Quorums{optHaveDIP0024Quorums.value_or(pindexPrev->nHeight >= consensusParams.DIP0024QuorumsHeight)};
     switch (llmqType)
     {
         case Consensus::LLMQType::LLMQ_DEVNET:
             return true;
         case Consensus::LLMQType::LLMQ_50_60:
-            if (Params().NetworkIDString() == CBaseChainParams::TESTNET) return true;
-            return !fDIP0024IsActive;
-
+            return !fDIP0024IsActive || !fHaveDIP0024Quorums ||
```
Comment (knst) [inline src/llmq/options.cpp:134]: this line should not be changed, should it?

Comment (UdjinM6) [reply src/llmq/options.cpp:134]: note `// fall through` here https://github.com/dashpay/dash/commit/b8a086d5e7d6c0307f429e3844e4331ffb036c89#diff-cf3d9716cca5e57a6033574399f5be8a184f017da2c02c414ae2771c101a2339R870

Comment (knst) [reply src/llmq/options.cpp:134]: yes, indeed, it has been changed in #5790 and reverted later in 5841 (this PR).
Both RRs are between v20.0.4 and v20.1, so, it should not affect then

**Pull request merged**